Lib/test/test_typing.py

```diff
@@ -5844,6 +5844,21 @@ class A:
                     with self.assertRaises(TypeError):
                         a[int]
 
+    def test_return_non_tuple_while_unpacking(self):
+        # GH-138497: GenericAlias objects didn't ensure that __typing_subst__ actually
+        # returned a tuple
+        class Generic:
+            def __call__(*_):
+                return None
+
+            def __getattr__(self, _):
+                return Generic()
+
+        evil = Generic()
```

Comment:
Generic conflicts with common Generic. This class is TypeVar-like, so let's call it EvilTypeVar.

It does not need to be callable -- only have some methods and attributes. Here is a minimal reproducer:


```suggestion
        class EvilTypeVar:
            __typing_is_unpacked_typevartuple__ = True
            def __typing_prepare_subst__(*_):
                return None  # any value
            def __typing_subst__(*_):
                return 42  # not tuple

        evil = EvilTypeVar()
```